src/runtime_src/tools/scripts/setup.zsh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2019-2021 Xilinx, Inc. All rights reserved.
+# Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Script to setup environment for XRT
+# This script is installed in /opt/xilinx/xrt and must
+# be sourced from that location
+
+# Check OS version requirement
+OSDIST=`cat /etc/os-release | grep -i "^ID=" | awk -F= '{print $2}'`
+if [[ $OSDIST == "centos" ]]; then
+    OSREL=`cat /etc/redhat-release | awk '{print $4}' | tr -d '"' | awk -F. '{print $1*100+$2}'`
+else
+    OSREL=`cat /etc/os-release | grep -i "^VERSION_ID=" | awk -F= '{print $2}' | tr -d '"' | awk -F. '{print $1*100+$2}'`
+fi
+
+if [[ $OSDIST == "ubuntu" ]]; then
+    if (( $OSREL < 1604 )); then
+        echo "ERROR: Ubuntu release version must be 16.04 or later"
+        return 1
+    fi
+fi
+
+if [[ $OSDIST == "centos" ]] || [[ $OSDIST == "rhel"* ]]; then
+    if (( $OSREL < 704 )); then
+        echo "ERROR: Centos or RHEL release version must be 7.4 or later"
+        return 1
+    fi
+fi
+
+XILINX_XRT=$(readlink -f $(dirname ${(%):-%N}))
+
+if [[ $XILINX_XRT != *"/opt/xilinx/xrt" ]]; then
+    echo "Invalid location: $XILINX_XRT"
+    echo "This script must be sourced from XRT install directory"
+    return 1
+fi
+
+COMP_FILE="/usr/share/bash-completion/bash_completion"
+# 1. This is a workaround for set -e
+# The issue is chaining conditionals with actual commands
+# The issue is caused when sourcing the ${COMP_FILE}.
+# Specifically ${COMP_FILE}::_sysvdirs. Each check in that function
+# will fail the script due to the issues.
+# If set -e is removed from the pipeline then check can be removed
+# 2. Make sure that the shell is bash! The completion may not function
+# correctly or setup on other shells.
+# 3. Make sure the bash completion file exists
+if [[ $- != *e* ]] && [[ "$BASH" == *"/bash" ]] && [ -f "${COMP_FILE}" ]; then
```

Comment:
I am a little bit lost here. I thought this was about `zsh`. Is there anything specific to `zsh` here anyway?